webapp/src/pages/OneTimePassword.js

```diff
@@ -56,12 +59,33 @@ const OneTimePassword = () => {
       })
       .catch((err) => {
         setOtp(new Array(6).fill(""));
+        setMessage(null);
         setError(extractErrorCode(err));
         const firstOtpField = document.getElementById("otpContainer").firstChild;
         firstOtpField.focus();
       });
   };
 
+  const handleResend = () => {
+    const firstOtpField = document.getElementById("otpContainer").firstChild;
+    setError(null);
+    api
+      .authStart({
+        phone: phoneNumber,
+        timezone: dayjs.tz.guess(),
+      })
+      .then(() => {
+        setOtp(new Array(6).fill(""));
```

Comment:
I'd move this entire handler to before the API call. Otherwise you click the button and nothing happens- we would have to introduce a loading state, but there's really no reason to do that since almost every time this will succeed.

Reply:
Exactly! Be optimistic in API calls where possible: assume success, and then handle the failure case.